include/utilities.h

```diff
@@ -459,4 +459,15 @@ void smart_delete (SmartPointer<TYPE> &sp)
     }
 }
 
+
+template <typename VEC>
+void vector_shift(VEC &in_vec, double a_scalar)
+{
+  for (auto i : in_vec.locally_owned_elements())
```

Comment:
both current versions of gcc and clang has full support for `auto`. deal.II during its configuration check the ability of the compiler to support c++11 and c++14. I guess that the deal2lkit users have a recent compiler that supports c++11. We can also add a guard or an Assert using `DEAL_II_WITH_CXX11` variable, which is automatically set during the configuration of deal.II